src/main/java/net/mcreator/workspace/elements/VariableElement.java

```diff
@@ -38,10 +43,32 @@
 	}
 
 	public VariableElementType getType() {
-		return type;
+		return getVariableFromType(type);
+	}
+
+	public static VariableElementType getVariableFromType(String type) {
+		for (VariableElementType varType : variables.keySet()) {
+			if(varType.getBlocklyVariableType().equalsIgnoreCase(type) || varType.getName().equalsIgnoreCase(type)
+					|| varType.getBlockName().equalsIgnoreCase(type)) {
+				return varType;
+			}
+		}
+		return null;
 	}
 
-	public void setType(VariableElementType type) {
+	public static Set<VariableElementType> getVariables() {
+		return variables.keySet();
+	}
+
+	public static void addVariable(VariableElementType var) {

```

Comment:
I don't understand what this does. addVariable but it takes variable type, what does this system do and why is it needed?

Reply:
This method is called every time a variable is loaded and can be used. It will add the variable gave in the `VARIABLE_CACHE` map variable. Then, this `VARIABLE_CACHE` map variable can use in the other getters of the class. I renamed it for `addVariableTypeToCache`.
It only takes a `VariableElementType` as a parameter because we can get its name directly with the VariableElementType.